src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/IssuedTokenParametersElement.cs

```diff
@@ -0,0 +1,328 @@
+                }
+
+                // No need to worry about a race condition here-- this method is not meant to be called by multi-threaded
+                // apps. It is only supposed to be called by svcutil and single threaded equivalents.
+                if (this.optionalIssuedTokenParameters == null)
+                {
+                    this.optionalIssuedTokenParameters = new Collection<IssuedTokenParametersElement>();
+                }
+                return this.optionalIssuedTokenParameters;
+            }
+        }
+
+
+        [ConfigurationProperty(ConfigurationStrings.TokenType, DefaultValue = "")]
+        [StringValidator(MinLength = 0)]
+        public string TokenType
+        {
+            get { return (string)base[ConfigurationStrings.TokenType]; }
+            set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    value = string.Empty;
+                }
+
+                base[ConfigurationStrings.TokenType] = value;
+            }
+        }
+
+        [ConfigurationProperty(ConfigurationStrings.UseStrTransform, DefaultValue = false)]
+        public bool UseStrTransform
+        {
+            get { return (bool)base[ConfigurationStrings.UseStrTransform]; }
+            set { base[ConfigurationStrings.UseStrTransform] = value; }
+        }
+
+        internal void ApplyConfiguration(IssuedSecurityTokenParameters parameters)
+        {
+            if (parameters == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("parameters"));
+
+            if (this.AdditionalRequestParameters != null)
+            {
+                foreach (XmlElementElement e in this.AdditionalRequestParameters)
+                {
+                    parameters.AdditionalRequestParameters.Add(e.XmlElement);
+                }
+            }
+
+            if (this.ClaimTypeRequirements != null)
+            {
+                foreach (ClaimTypeElement c in this.ClaimTypeRequirements)
+                {
+                    parameters.ClaimTypeRequirements.Add(new ClaimTypeRequirement(c.ClaimType, c.IsOptional));
+                }
+            }
+
+            parameters.KeySize = this.KeySize;
+            //TODO: This is an internal property
+            //parameters.KeyType = this.KeyType;
+            parameters.DefaultMessageSecurityVersion = this.DefaultMessageSecurityVersion;
+            parameters.UseStrTransform = this.UseStrTransform;
+
+            if (!string.IsNullOrEmpty(this.TokenType))
+            {
+                parameters.TokenType = this.TokenType;
+            }
+            if (PropertyValueOrigin.Default != this.ElementInformation.Properties[ConfigurationStrings.Issuer].ValueOrigin)
+            {
+                //TODO: Implement IssuedTokenParameters
+                //this.Issuer.Validate();
```

Comment:
Can you throw a PlatformNotSupportedException here? If they have an issuer explicitly set, and we don't use it, that could result in a hard to track down bug. This code block is protected against the default case so the exception would only throw when really needed.

Reply:
I have uncommented these